### PR TITLE
Update the NRF lock implementation to use the Door Lock State

### LINKED ISSUE #16889: NRFConnect Lock App is using the OnOff Cluster
#### Problem
The Nordic lock example implementation is using the OnOff cluster to set Lock State. If the door is unlocked, the OnOff Cluster will be set to false indicating to controllers that the accessory is not on. 

 #### Proposed Solution
Nordic implementation of the Lock needs to be updated to use the Lock Cluster.

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
* NRF Lock implementation is using the on/off cluster for lock state machine.
* Fixes #16889 

#### Change overview
* Update the NRF lock implementation to use the Door Lock State attribute for lock state machine.

#### Testing
* Generated code
* Compiled
* Ran chip-tool lock and unlock commands
* Tested Subscribe 
* Tested Button 2

### CODE REVIEW

**Comment by kkasperczyk-no** [comment]
@krypton36 thank you for contribution to nrfconnect example! We already planned to do such change for lock-app example, but it is very nice to see the help from you. 

I think there was discussion on software development Slack channel with @tcarmelveilleux, @LuDuda and @bzbarsky-apple  about lock-app vs door-lock-app and the conclusion was that door-lock should not be added as a new example at all, but we should align existing lock-app to the door lock device type requirements. Would you mind trying to modify lock-app example with your changes instead of creating new nrfconnect implementation of door-lock-app?

**Review by msandstedt — APPROVED** [review]

File: examples/lock-app/nrfconnect/README.md [hunk 1]
```diff
@@ -391,11 +391,11 @@ To build for the low-power configuration, run the following command with
 _build-target_ replaced with the build target name of the Nordic Semiconductor's
 kit you own (for example `nrf52840dk_nrf52840`):
 
-    $ west build -b build-target -- -DOVERLAY_CONFIG=overlay-low_power.conf
+    $ west build -b build-target -- -DOVERLAY_CONFIG=../../overlay-low_power.conf
```
Comment (Damian-Nordic) [inline examples/lock-app/nrfconnect/README.md:394]: That looks like unintentional change. It was changed back and forth recently and you probably so you probably used the intermediate commit as a base for your changes.

File: examples/lock-app/nrfconnect/main/AppTask.cpp [hunk 2]
```diff
@@ -549,11 +550,18 @@ void AppTask::DispatchEvent(AppEvent * aEvent)
 
 void AppTask::UpdateClusterState()
 {
-    // write the new on/off value
-    EmberAfStatus status = Clusters::OnOff::Attributes::OnOff::Set(kLockEndpointId, !BoltLockMgr().IsUnlocked());
-
+    EmberAfStatus status;
+    LOG_INF("Updating door lock state");
+    chip::app::DataModel::Nullable<DlLockState> lockSate;
+    status = Clusters::DoorLock::Attributes::LockState::Get(kLockEndpointId, lockSate);
```
Comment (Damian-Nordic) [inline examples/lock-app/nrfconnect/main/AppTask.cpp:556]: The purpose of this function was to synchronize the cluster state with `BoltLockMgr().IsUnlock()`. The latter could be manipulated with a Button. This code, however, seems to toggle the state, which I'm not sure is 100% accurate.

**Review by Damian-Nordic — COMMENTED** [review]
Thanks for contributing. I have a few questions though.

File: examples/lock-app/nrfconnect/README.md (hunk 1 above)
Comment (krypton36) [reply examples/lock-app/nrfconnect/README.md:394]: Definitely unintended. I will update.

File: examples/lock-app/nrfconnect/main/AppTask.cpp (hunk 2 above)
Comment (krypton36) [reply examples/lock-app/nrfconnect/main/AppTask.cpp:556]: You make a good point. I was running into issue, where the BoltLockMgr state would not match the server state and the button would not change the state. I will update this PR to use the callbacks below to update BoltLockMgr state instead.

File: examples/lock-app/nrfconnect/main/ZclCallbacks.cpp [hunk 3]
```diff
@@ -19,39 +19,56 @@
...
-void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath & attributePath, uint8_t mask, uint8_t type,
-                                       uint16_t size, uint8_t * value)
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
+
+bool emberAfPluginDoorLockOnDoorLockCommand(chip::EndpointId endpointId, const Optional<ByteSpan> & pinCode, DlOperationError & err)
 {
-    if (attributePath.mClusterId == OnOff::Id && attributePath.mAttributeId == OnOff::Attributes::OnOff::Id)
+    EmberAfStatus status;
+    LOG_INF("Locking");
+    status = Attributes::LockState::Set(endpointId, DlLockState::kLocked);
```
[the remark this answers is not in the document]
Comment (krypton36) [reply examples/lock-app/nrfconnect/main/ZclCallbacks.cpp:37]: Definitely not needed. Just looked at door-lock-server. Thank you. I will remove and update the BoltLockMgr state here.

File: examples/lock-app/nrfconnect/README.md (hunk 1 above)
Comment (krypton36) [reply examples/lock-app/nrfconnect/README.md]: Fixed.

File: examples/lock-app/nrfconnect/main/AppTask.cpp (hunk 2 above)
Comment (krypton36) [reply examples/lock-app/nrfconnect/main/AppTask.cpp]: Updated and retested.

File: examples/lock-app/nrfconnect/main/ZclCallbacks.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (krypton36) [reply examples/lock-app/nrfconnect/main/ZclCallbacks.cpp]: Updated and retested.

**Review by Damian-Nordic — APPROVED** [review]

**Comment by Damian-Nordic** [comment]
@krypton36 I rebased your change and did some CI fixes as I need to push another change that depends on yours :)

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**